Framework/Algorithms/src/FindPeaks.cpp

```diff
@@ -462,8 +464,12 @@ void FindPeaks::findPeaksUsingMariscotti() {
   const int blocksize = static_cast<int>(smoothedData->blocksize());
 
   for (int k = start; k < end; ++k) {
-    const auto &S = smoothedData->y(k);
-    const auto &F = smoothedData->e(k);
+    int ws_i(k);
+    if (singleSpectrum)
+      ws_i = 0;
```

Comment:
Why do you need these changes, aren't `start` and `end` above set up correctly to do this already?